here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/apis/ApiParams.java

```diff
@@ -18,20 +18,101 @@
  */
 package com.here.naksha.app.service.http.apis;
 
+import com.here.naksha.lib.core.exceptions.XyzErrorException;
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.payload.events.QueryParameter;
+import com.here.naksha.lib.core.models.payload.events.QueryParameterList;
+import com.here.naksha.lib.core.util.ValueList;
 import io.vertx.ext.web.RoutingContext;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public final class ApiParams {
+  public static String STORAGE_ID = "storageId";
   public static String SPACE_ID = "spaceId";
   public static String PREFIX_ID = "prefixId";
   public static String FEATURE_ID = "featureId";
   public static String ADD_TAGS = "addTags";
   public static String REMOVE_TAGS = "removeTags";
+  public static String TAGS = "tags";
   public static String FEATURE_IDS = "id";
+  public static String WEST = "west";
+  public static String NORTH = "north";
+  public static String EAST = "east";
+  public static String SOUTH = "south";
+  public static String LIMIT = "limit";
+  public static String TILE_TYPE = "type";
+  public static String TILE_ID = "tileId";
 
-  public static @Nullable String pathParam(
+  public static long DEF_FEATURE_LIMIT = 30_000;
+
+  public static @NotNull String extractMandatoryPathParam(
       final @NotNull RoutingContext routingContext, final @NotNull String param) {
-    return routingContext.pathParam(param);
+    final String value = routingContext.pathParam(param);
+    if (value == null || value.isEmpty()) {
+      throw new XyzErrorException(XyzError.ILLEGAL_ARGUMENT, "Missing " + param + " parameter");
+    }
+    return value;
+  }
+
+  public static double extractQueryParamAsDouble(
+      final @NotNull QueryParameterList queryParams, final @NotNull String key, final boolean isMandatory) {
+    return extractQueryParamAsDouble(queryParams, key, isMandatory, 0.0);
+  }
+
+  public static double extractQueryParamAsDouble(
+      final @NotNull QueryParameterList queryParams,
```

Comment:
since the isMandatory logic is there, I would prefer marking `queryParams` as nullable